src/playbacks/hls/hls.js

```diff
@@ -28,12 +29,73 @@ export default class HLS extends HTML5VideoPlayback {
+    }
+    return this._playableRegionStartTime
+  }
+
+  get _now() {
+    if (window.performance && window.performance.now) {
+      return performance.now()
+    }
+    return Date.now()
+  }
+
+  // the time in the video element which should represent the start of the sliding window
+  // extrapolated to increase in real time (instead of jumping as the early segments are removed)
+  get _extrapolatedStartTime() {
+    if (!this._localStartTimeCorrelation) {
+      return this._playableRegionStartTime
+    }
+    var corr = this._localStartTimeCorrelation
+    var timePassed = this._now - corr.local
+    var extrapolatedWindowStartTime = (corr.remote + timePassed) / 1000
+    // cap at the end of the extrapolated window duration
+    return Math.min(extrapolatedWindowStartTime, this._playableRegionStartTime + this._extrapolatedWindowDuration)
+  }
+
+  // Returns the duration (seconds) of the window that the extrapolated start time is allowed
+  // to move in before being capped.
+  // The extrapolated start time should never reach the cap at the end of the window as the
+  // window should slide as chunks are removed from the start.
+  //
+  // If chunks aren't being removed for some reason that the start time will reach and remain fixed at
+  // playableRegionStartTime + extrapolatedWindowDuration
+  //
+  //                                <-- window duration --> 
+  // I.e   playableRegionStartTime |-----------------------|
+  //                               | -->   .       .       .
+  //                               .   --> | -->   .       .
+  //                               .       .   --> | -->   .
+  //                               .       .       .   --> |
+  //                               .       .       .       .
+  //                                 extrapolatedStartTime                 
+  get _extrapolatedWindowDuration() {
+    if (this._segmentTargetDuration === null) {
+      return 0
+    }
+    return this._extrapolatedWindowNumSegments * this._segmentTargetDuration
+  }
+
   constructor(...args) {
     super(...args)
     // backwards compatibility (TODO: remove on 0.3.0)
     this.options.playback || (this.options.playback = this.options.hlsjsConfig)
-    this._minDvrSize = (this.options.hlsMinimumDvrSize === undefined) ? 60 : this.options.hlsMinimumDvrSize
+    this._minDvrSize = typeof(this.options.hlsMinimumDvrSize) === "undefined" ? 60 : this.options.hlsMinimumDvrSize
+    // The size of the start time extrapolation window measured as a multiple of segments.
+    // Should be 2 or higher, or 0 to disable. Should only need to be increased above 2 if more than one segment is
+    // removed from the start of the playlist at a time. E.g if the playlist is cached for 10 seconds and new chunks are
+    // added/removed every 5.
+    this._extrapolatedWindowNumSegments = typeof(this.options.playback.extrapolatedWindowNumSegments) === "undefined" ? 2 :  this.options.playback.extrapolatedWindowNumSegments
```

Comment:
@leandromoreira as we discussed before, we should stick to `options.playback` for playback-specific options, and `options.pluginName` for core/container plugin-specific options.

I don't think we really need to have options separated by playback name, because this way we can use the same option name/value for any playback used for a given source. For example, we could use the same option name for hls.js and FlasHLS playbacks.